token-swap/program/src/processor.rs

```diff
@@ -672,6 +676,15 @@ impl PrintProgramError for SwapError {
     }
 }
 
+fn to_u128(val: u64) -> Result<u128, SwapError> {
+    val.try_into().map_err(|_| SwapError::CalculationFailure)
+}
+
+fn to_u64(val: u128) -> Result<u64, SwapError> {
+    val.try_into().map_err(|_| SwapError::CalculationFailure)
```

Comment:
A different error like `OverflowError` would make things clearer.  Could anything else make this fail?